rbac/rbac.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package rbac
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+type scopeContextKeyType int
+
+const (
+	scopeContextKey scopeContextKeyType = 0
```

Comment:
How does this work? Are we sure it won't clash with anything else?

Reply:
that's a tricky one;
what you see here is a common pattern, because the combination of type and value has to be unique;
this is why we have scopeContextKeyType instead of simple int; this type should be different for each package
golang uses empty interfaces for context keys and according to docs
_Two interface values are equal if they have identical dynamic types and equal dynamic values or if both have value nil._